### PR TITLE
remove --ssh- options, deprecated 13 releases, that only work on GCE

### PR DESCRIPTION
**Pull request by deads2k**
Removing --ssh-user and --ssh-key options that have been deprecated for 13 releases (since 1.9).  They only ever functioned on GCE and now only function when running the in-tree cloud provider instead of the out of the out of tree provider.

The apiserver proxy feature was introduced in 1.16 and went to beta in 1.18, so a replacement has been available for some time.

I bumped into this while trying to fix an integration test that needed fine control of some versioning options and I wasn't able to unwind it in part because of the construction of the tunneler.

PR opened to see if trip dependencies in CI have been built on this.

/kind cleanup
/priority important-soon
@kubernetes/sig-api-machinery-api-reviews 

```release-note
--ssh-user and --ssh-key options are removed.  They only functioned on GCE, and only in-tree.  Use the apiserver network proxy instead.
```

### CODE REVIEW

**Comment by liggitt** [comment]
just to make sure all network proxy paths remain functional as this is dropped

/test pull-kubernetes-e2e-gce-network-proxy-grpc

**Comment by fedebongio** [comment]
yay!

**Comment by cheftako** [comment]
/lgtm

https://testgrid.k8s.io/sig-api-machinery-network-proxy#ci-kubernetes-e2e-gci-gce-network-proxy-grpc&width=25 (the ssh tunnel replacement looks green) and https://github.com/kubernetes/kubernetes/pull/102510 has it being tested. I think we are good to go. Thanks for getting the removal handled.

**Comment by deads2k** [comment]
/test all

just ensuring we're still green.

File: cluster/gce/gci/configure-kubeapiserver.sh [hunk 1]
```diff
@@ -276,23 +276,12 @@ function start-kube-apiserver {
...
-      params+=" --ssh-user=${PROXY_SSH_USER}"
-      params+=" --ssh-keyfile=/etc/srv/sshproxy/.sshkeyfile"
-    fi
   elif [[ -n "${PROJECT_ID:-}" && -n "${TOKEN_URL:-}" && -n "${TOKEN_BODY:-}" && -n "${NODE_NETWORK:-}" ]]; then
     local -r vm_external_ip=$(get-metadata-value "instance/network-interfaces/0/access-configs/0/external-ip")
-    if [[ -n "${PROXY_SSH_USER:-}" ]]; then
-      if [[ -n "${KUBE_API_SERVER_RUNASUSER:-}" && -n "${KUBE_API_SERVER_RUNASGROUP:-}" ]]; then
-        chown -R "${KUBE_API_SERVER_RUNASUSER}":"${KUBE_API_SERVER_RUNASGROUP}" /etc/srv/sshproxy/
-      fi
-      params+=" --advertise-address=${vm_external_ip}"
```
Comment (liggitt) [inline cluster/gce/gci/configure-kubeapiserver.sh:292]: this is a weird line to have gotten stuck in the PROXY_SSH_USER block... I wonder if that was an accident

File: cluster/gce/gci/configure-kubeapiserver.sh [hunk 2]
```diff
@@ -276,23 +276,12 @@ function start-kube-apiserver {
...
-      params+=" --ssh-user=${PROXY_SSH_USER}"
-      params+=" --ssh-keyfile=/etc/srv/sshproxy/.sshkeyfile"
+    if [[ -n "${KUBE_API_SERVER_RUNASUSER:-}" && -n "${KUBE_API_SERVER_RUNASGROUP:-}" ]]; then
+      chown -R "${KUBE_API_SERVER_RUNASUSER}":"${KUBE_API_SERVER_RUNASGROUP}" /etc/srv/sshproxy/
     fi
+    params+=" --advertise-address=${vm_external_ip}"
```
Comment (cheftako) [inline cluster/gce/gci/configure-kubeapiserver.sh:284]: Sorry to nit but can we move this up to be just after where we set the vm_external_ip variable?

Comment (deads2k) [reply cluster/gce/gci/configure-kubeapiserver.sh:284]: > Sorry to nit but can we move this up to be just after where we set the vm_external_ip variable?

sure, done.

File: cmd/kube-apiserver/app/server.go [hunk 3]
```diff
@@ -55,7 +54,6 @@ import (
 	clientgoclientset "k8s.io/client-go/kubernetes"
 	"k8s.io/client-go/rest"
 	"k8s.io/client-go/util/keyutil"
-	cloudprovider "k8s.io/cloud-provider"
```
Comment (cheftako) [inline cmd/kube-apiserver/app/server.go:58]: Woohoo!

**Comment by andrewsykim** [comment]
Really great to see this happen, thanks for all the effort folks!

**Comment by dims** [comment]
image

**Comment by dims** [comment]
DGEoPl3

**Comment by cheftako** [comment]
:stuck_out_tongue_closed_eyes:

**Comment by cheftako** [comment]
Congrats! Thanks everyone for making this happen!

**Pull request merged**